extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/DifferenceWithinRangeScrutinizer.java

```diff
@@ -0,0 +1,53 @@
+package org.openrefine.wikidata.qa.scrutinizers;
+
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.wikidata.wdtk.datamodel.interfaces.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DifferenceWithinRangeScrutinizer extends EditScrutinizer {
+
+    public static final String type = "difference-of-the-properties-is-not-within-the-specified-range";
+
+    @Override
+    public void scrutinize(ItemUpdate update) {
+        Map<PropertyIdValue, Value> propertyIdValueValueMap = new HashMap<>();
+        for (Statement statement : update.getAddedStatements()){
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            Value value = statement.getClaim().getMainSnak().getValue();
+            propertyIdValueValueMap.put(pid, value);
+        }
+
+        for(PropertyIdValue propertyId : propertyIdValueValueMap.keySet()){
+            if (_fetcher.hasDiffWithinRange(propertyId)){
+                PropertyIdValue lowerPropertyId = _fetcher.getLowerPropertyId(propertyId);
+                QuantityValue minRangeValue = _fetcher.getMinimumValue(propertyId);
+                QuantityValue maxRangeValue = _fetcher.getMaximumValue(propertyId);
```

Comment:
We should handle the case where only some of these parameters have been defined: we should make sure we do not fail with a null pointer exception, for instance…

Reply:
When I went through the docs It was mentioned that these are [mandatory parameters](https://www.wikidata.org/wiki/Help:Property_constraints_portal/Diff_within_range) for the constraint. Therefore, I didn't add the conditions. If you say so, I'll add those checks.

Reply:
They are mandatory indeed, but there is nothing preventing users from not adding them to the constraint definition.

Reply:
Done!